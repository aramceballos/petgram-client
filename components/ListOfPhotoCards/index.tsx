import React from 'react'
import Skeleton from '@material-ui/lab/Skeleton'

import PhotoCard from '../PhotoCard'
import { List } from './styles'

type Props = {
  posts: IPost[]
  loading?: boolean
}

const ListOfPhotoCardsComponent = ({ posts, loading }: Props) => {
  if (loading) {
    return (
      <div>
        <div style={{ display: 'flex', marginTop: '35px' }}>
          <Skeleton
            style={{ marginLeft: '15px' }}
            animation="wave"
            variant="circle"
            width={30}
            height={30}
          />
          <Skeleton
            style={{ marginLeft: '15px' }}
            animation="wave"
            variant="text"
            width="80%"
          />
        </div>
        <Skeleton
          style={{ marginTop: '15px' }}
          animation="wave"
          variant="rect"
          height={200}
        />
        <div style={{ display: 'flex', marginTop: '35px' }}>
          <Skeleton
            style={{ marginLeft: '15px' }}
            animation="wave"
            variant="circle"
            width={30}
            height={30}
          />
          <Skeleton
            style={{ marginLeft: '15px' }}
            animation="wave"
            variant="text"
            width="80%"
          />
        </div>
        <Skeleton
          style={{ marginTop: '15px' }}
          animation="wave"
          variant="rect"
          height={200}
        />
      </div>
    )
  }

  return (
    <List>
      {posts && posts.map((post) => <PhotoCard key={post.id} {...post} />)}
    </List>
  )
}

export default ListOfPhotoCardsComponent
