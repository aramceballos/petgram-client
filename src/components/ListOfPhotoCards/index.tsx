import React from 'react'

import { List } from './styles'
import Photocard from '../PhotoCard'

type Props = {
  posts: IPost[]
}

const ListOfPhotoCardsComponent = ({ posts = [] }: Props) => {
  return (
    <List>
      {posts.map((post) => (
        <Photocard key={post.id} {...post} />
      ))}
    </List>
  )
}

export default ListOfPhotoCardsComponent
