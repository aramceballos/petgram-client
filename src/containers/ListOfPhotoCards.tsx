import React from 'react'

import ListOfPhotoCardsComponent from '../components/ListOfPhotoCards'
import { useGetRequest } from '../hooks/useRequest'

const ListOfPhotoCards = () => {
  const { data: posts, loading } = useGetRequest('/p')

  return <ListOfPhotoCardsComponent posts={posts} loading={loading} />
}

export default ListOfPhotoCards
