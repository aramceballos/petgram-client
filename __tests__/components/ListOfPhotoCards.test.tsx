import React from 'react'
import { shallow } from 'enzyme'
import ListOfPhotoCards from '../../components/ListOfPhotoCards'

describe('<ListOfPhotoCards />', () => {
  test('should render correctly with no data', () => {
    const props = {
      posts: [],
      loading: false,
    }
    const component = shallow(<ListOfPhotoCards {...props} />)
    expect(component).toMatchSnapshot()
  })

  test('should render correctly with data', () => {
    const props = {
      posts: [
        {
          id: 5,
          user_id: 3,
          category_id: 50,
          post_date: '2021-05-25 05:38:27.151523',
          image_url:
            'https://res.cloudinary.com/midudev/image/upload/w_300/q_80/v1560262103/dogs.png',
          description: 'testing description',
          name: 'testing name',
          username: 'testing.username',
          email: 'test@test.com',
          likes: [
            {
              id: 2,
              post_id: 23,
              user_id: 45,
            },
            {
              id: 456,
              post_id: 6854,
              user_id: 4876,
            },
          ],
        },
        {
          id: 4879,
          user_id: 489,
          category_id: 489,
          post_date: '2021-05-25 05:38:27.151523',
          image_url:
            'https://res.cloudinary.com/midudev/image/upload/w_300/q_80/v1560262103/dogs.png',
          description: 'another description',
          name: 'another name',
          username: 'another.username',
          email: 'test@test.com',
          likes: [
            {
              id: 2,
              post_id: 23,
              user_id: 45,
            },
            {
              id: 456,
              post_id: 6854,
              user_id: 4876,
            },
          ],
        },
      ],
      loading: false,
    }
    const component = shallow(<ListOfPhotoCards {...props} />)
    expect(component).toMatchSnapshot()
  })
})
