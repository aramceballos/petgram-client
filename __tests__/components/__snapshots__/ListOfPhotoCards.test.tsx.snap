// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListOfPhotoCards /> should render correctly with data 1`] = `ShallowWrapper {}`;

exports[`<ListOfPhotoCards /> should render correctly with no data 1`] = `ShallowWrapper {}`;
